ssms.go

```diff
@@ -57,15 +57,10 @@ func (sm *SSMuxer) SecureOutbound(ctx context.Context, insecure net.Conn, p peer
 
 	var sconn sec.SecureConn
 	if server {
-		sconn, err = tpt.SecureInbound(ctx, insecure)
+		sconn, err = tpt.SecureInbound(ctx, insecure, p)
 		if err != nil {
 			return nil, false, fmt.Errorf("failed to secure inbound connection: %s", err)
 		}
-		// ensure the correct peer connected to us
-		if sconn.RemotePeer() != p {
```

Comment:
What about `p != "" && p != sconn.RemotePeer()`? Just slightly safer.

Reply:
Why? In case our secure transports mess up?
Should we make this an assertion then (i.e. a `panic`)?

Reply:
Yes. I'd log loudly and fail, but I wouldn't panic (would turn a bug into a DoS). Again, we should catch this at a higher layer as well, but this is something that's nice to check at all layers just in case we mess something up in one of them.